app/controllers/book_tours_controller.rb

```diff
@@ -1,2 +1,47 @@
 class BookToursController < ApplicationController
+  before_action :logged_in_user, only: %i(index create)
+  before_action :list_book_tour, only: %i(index)
+  before_action :find_tour, only: %i(create)
+
+  def index; end
+
+  def create
+    @book_tour = current_user.book_tours.new book_tour_params
+    @book_tour.price_tour = total_price
+    respond_to do |format|
+      if @book_tour.save
+        format.html{redirect_to @book_tour, message: "thanh cong"}
+      else
+        format.html{render :new}
+      end
+      format.js {}
+    end
+  end
+
+  private
+
+  def find_tour
+    @tour = Tour.find_by(id: params[:book_tour][:tour_id])
+
+    return if @tour
+
+    flash[:error] = t "not_found_tour"
+    redirect_to root_path
+  end
+
+  def book_tour_params
+    params.require(:book_tour).permit :tour_id,
+                                      :quantity, :start_date, :end_date,
+                                      :customer_name, :phone_number, :email
+  end
+
+  def list_book_tour
```

Comment:
Đặt tên thế này thôi e 
```suggestion
  def load_user_book_tours
```